### DISCUSSION TITLE
How to use programmatically when gallery can't be modified?

### DISCUSSION THREAD

**Discussion by johtso** [dromru/react-photoswipe-gallery#819] [Q&A] [answered]
I'm wondering how you would use this library to open a specific image in the lightbox without having to use `Gallery` and `Item`.

I have an existing gallery using the masonic library for virtualised infinite scrolling.

What I need is a callback that I can bind to the onclick of my images, and some way to give photoswipe an array of the images in my gallery.

**Comment by irudoy** [comment, marked as answer]
This wrapper aims to serialize React elements tree into the plain array and provide it to the PhotoSwipe. Seems like it's better to initialize PhotoSwipe manually in the `useEffect` hook for your particular case. [This example](https://photoswipe.com/react-image-gallery/) covers it. But I would suggest using the `PhotoSwipe` class instead of `PhotoSwipeLightbox` for a more flexible configuration.

Anyway, if you think this library can have some API that could reduce configuration complexity – feel free to create an issue with a proposal and a clear explanation of which problem it solves.

**Reply by johtso** [reply]
Thanks, I thought I'd probably have to go that route! I'll see how it looks when I'm done and whether it would make any sense to have some abstraction for it
